From 2757ad843d7e6f18ae1994461c9a97a987b3663e Mon Sep 17 00:00:00 2001
From: Bernd Geiser <bg@ferncast.de>
Date: Thu, 29 Jun 2023 15:56:36 +0200
Subject: [PATCH] uri-asset-use-custom-context-for-running-discoverer-

---
 ges/ges-uri-asset.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ges/ges-uri-asset.c b/ges/ges-uri-asset.c
index 9dd576d..65be0ed 100644
--- a/ges/ges-uri-asset.c
+++ b/ges/ges-uri-asset.c
@@ -702,6 +702,7 @@ ges_uri_clip_asset_request_sync (const gchar * uri, GError ** error)
   GESUriClipAsset *asset;
   RequestSyncData data = { 0, };
   GstDiscoverer *previous_discoverer;
+  GMainContext *ctx;
 
   asset = GES_URI_CLIP_ASSET (ges_asset_request (GES_TYPE_URI_CLIP, uri,
           &lerror));
@@ -709,7 +710,9 @@ ges_uri_clip_asset_request_sync (const gchar * uri, GError ** error)
   if (asset)
     return asset;
 
-  data.ml = g_main_loop_new (NULL, TRUE);
+  ctx = g_main_context_new();
+  g_main_context_push_thread_default (ctx);
+  data.ml = g_main_loop_new (ctx, TRUE);
   previous_discoverer = get_discoverer ();
   create_discoverer ();
 
@@ -717,6 +720,8 @@ ges_uri_clip_asset_request_sync (const gchar * uri, GError ** error)
       (GAsyncReadyCallback) asset_ready_cb, &data);
   g_main_loop_run (data.ml);
   g_main_loop_unref (data.ml);
+  g_main_context_pop_thread_default (ctx);
+  g_main_context_unref (ctx);
 
   G_LOCK (discoverers_lock);
   g_hash_table_insert (discoverers, g_thread_self (), previous_discoverer);
-- 
2.39.2

