From 83ee983d74825af5d223cd5af0a32e2c0dc72f6d Mon Sep 17 00:00:00 2001
From: Bernd Geiser <bg@ferncast.de>
Date: Fri, 22 Sep 2023 16:11:05 +0200
Subject: [PATCH 4/6] ges-uri-asset-disable-cached-discoverers

---
 ges/ges-uri-asset.c | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/ges/ges-uri-asset.c b/ges/ges-uri-asset.c
index 65be0ed..446a1d1 100644
--- a/ges/ges-uri-asset.c
+++ b/ges/ges-uri-asset.c
@@ -167,7 +167,12 @@ _start_loading (GESAsset * asset, GError ** error)
 {
   gboolean ret;
   const gchar *uri;
-  GstDiscoverer *discoverer = get_discoverer ();
+  //  GstDiscoverer *discoverer = get_discoverer ();
+
+  GstDiscoverer *discoverer = gst_discoverer_new (discovering_timeout, NULL);
+  g_signal_connect (discoverer, "discovered", G_CALLBACK (discoverer_discovered_cb),
+      NULL);
+  gst_discoverer_start (discoverer);
 
   uri = ges_asset_get_id (asset);
   GST_DEBUG_OBJECT (discoverer, "Started loading %s", uri);
@@ -701,7 +706,7 @@ ges_uri_clip_asset_request_sync (const gchar * uri, GError ** error)
   GError *lerror = NULL;
   GESUriClipAsset *asset;
   RequestSyncData data = { 0, };
-  GstDiscoverer *previous_discoverer;
+    /* GstDiscoverer *previous_discoverer; */
   GMainContext *ctx;
 
   asset = GES_URI_CLIP_ASSET (ges_asset_request (GES_TYPE_URI_CLIP, uri,
@@ -713,8 +718,8 @@ ges_uri_clip_asset_request_sync (const gchar * uri, GError ** error)
   ctx = g_main_context_new();
   g_main_context_push_thread_default (ctx);
   data.ml = g_main_loop_new (ctx, TRUE);
-  previous_discoverer = get_discoverer ();
-  create_discoverer ();
+  /* previous_discoverer = get_discoverer (); */
+  /* create_discoverer (); */
 
   ges_asset_request_async (GES_TYPE_URI_CLIP, uri, NULL,
       (GAsyncReadyCallback) asset_ready_cb, &data);
@@ -723,9 +728,9 @@ ges_uri_clip_asset_request_sync (const gchar * uri, GError ** error)
   g_main_context_pop_thread_default (ctx);
   g_main_context_unref (ctx);
 
-  G_LOCK (discoverers_lock);
-  g_hash_table_insert (discoverers, g_thread_self (), previous_discoverer);
-  G_UNLOCK (discoverers_lock);
+  /* G_LOCK (discoverers_lock); */
+  /* g_hash_table_insert (discoverers, g_thread_self (), previous_discoverer); */
+  /* G_UNLOCK (discoverers_lock); */
 
   if (data.error) {
     GST_ERROR ("Got an error requesting asset: %s", data.error->message);
-- 
2.39.2

